packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -92,6 +92,10 @@ class BoxDecoration extends Decoration {
     this.boxShadow,
     this.gradient,
     this.backgroundBlendMode,
+    @Deprecated(
+      'Replace BoxDecoration(shape: BoxShape.circle) with ShapeDecoration(shape: const CircleBorder() '
+      'This feature was deprecated after [???]',
```

Comment:
The deprecation policy has guidance on how to assign this version number: https://github.com/flutter/flutter/wiki/Tree-hygiene#deprecations